support-workers/src/main/scala/com/gu/support/workers/lambdas/SendThankYouEmail.scala

```diff
@@ -44,19 +44,32 @@ class SendThankYouEmail(
             response <- services.paperRoundService.agents()
           } yield response.data.agents.find(_.refId == id)
       }
+
     val emailBuilder = new EmailBuilder(
       services.zuoraService.getMandateIdFromAccountNumber,
       getAgentDetails,
       services.promotionService,
       Configuration.stage,
     )
 
-    for {
-      emailFields <- emailBuilder.buildEmail(state)
-      emailResult <- Future.sequence(emailFields.map(emailService.send))
-    } yield HandlerResult(emailResult, requestInfo)
+    /** We generate these users `testGuardianDoNotUse` as part of our e2e testing and don't want to create a braze user
+      * for them as if the emails bounce it might affect our reputation with email clients in the long run.
+      *
+      * see: support-e2e/tests/utils/users.ts
+      *
+      * We cannot use `isTestUser` as we do actually use this for testing receiving of emails and it _might_ still have
+      * a negative affect on the reputation as email clients are not aware it is a test account.
+      */
+    val email = state.user.primaryEmailAddress
+    if (email.startsWith("test.e2e.supporter.revenue") && email.endsWith("theguardian.com")) {
```

Comment:
Would it be worth extracting this pattern/prefix out to a val somewhere so it's used both by the check and the code which generates the email address? That way it'd be impossible for them to get out of sync.

Reply:
I've linked to the file above and can't think of a way of sharing that between projects that doesn't add a tonne of complexity that hopefully the comment mostly addresses. 

To this end, I have added a comment to the `users.ts` file too.